cg/store/api/add.py

```diff
@@ -204,14 +204,7 @@ def add_case(
     ) -> Case:
         """Build a new Case record."""
 
-        # generate a unique case id
-        while True:
-            internal_id = petname.Generate(2, separator="")
-            if self.get_case_by_internal_id(internal_id) is None:
-                break
-            else:
-                LOG.debug(f"{internal_id} already used - trying another id")
-
+        internal_id: str = self.generate_unique_petname()
```

Comment:
I think this function is used for samples currently and introduces the possibility that cases share internal ids with other cases. This would probably mean a crash at a later stage due to integrity errors when adding to the DB.

I think we should have a different method for generating petnames for samples and cases (either we change some parameter or use some separator, e.g. underscores or use some other way to assign petnames to samples). If we use this method, we should have a check for clashing ids with other cases though.

Reply:
Ah, yeah - I missed that the function makes the check against tha sample table 👍